### PR TITLE
#35 Added async variants for table members

### LINKED ISSUE #35: Async support for Tables
Queries on table storage do not support Async operations currently.

### PR DESCRIPTION
**Pull request by stewart-r**
Fixes #35 

I think I have now covered all the table members with async variants. 

As always, just let me know if there is something I ought to amend or rethink.

Stewart

### CODE REVIEW

**Comment by isaacabraham** [comment]
Thanks again :-)

**Comment by stewart-r** [comment]
My pleasure. (literally! ;-))

File: src/FSharp.Azure.StorageTypeProvider/Table/TableRepository.fs [hunk 1]
```diff
@@ -62,19 +64,46 @@ let internal getRowsForSchema (rowCount: int) connection tableName =
...
+                    |> Seq.map(fun p -> p.Key, p.Value.PropertyAsObject)
+                    |> Map.ofSeq)
+
+let executeGenericQueryAsync<'a> connection tableName maxResults filterString mapToReturnEntity = async{
+    let query = DynamicQuery().Where(filterString)
+    let query = if maxResults > 0 then query.Take(Nullable maxResults) else query
+        
+    let resSet = ResizeArray<'a>()
+    let getBatch contTkn = async {
+        let! batch = (getTable tableName connection).ExecuteQuerySegmentedAsync(query,contTkn) |> Async.AwaitTask
```
Comment (isaacabraham) [inline src/FSharp.Azure.StorageTypeProvider/Table/TableRepository.fs]: Where does the "new" contTkn get set?

**Comment by isaacabraham** [comment]
@stewart-r just a small question regarding the above - I'm thinking of rewriting this bit recursively / with a fold but I'm unclear of the behaviour of the continuation token. Does it mutate the exist token (in which case does it really set it to null?) or should there be a "new" token passed out?

**Comment by stewart-r** [comment]
Oh no - that's a mistake. 

`getBatch` should return` batch.ContinuationToken` - not the original `contTkn`:

```
let getBatch contTkn = async {
        let! batch = (getTable tableName connection).ExecuteQuerySegmentedAsync(query,contTkn) |> Async.AwaitTask
        batch 
        |> Seq.map(mapToReturnEntity)
        |> resSet.AddRange
        return batch.ContinuationToken
    } 
    let! firstContTkn = getBatch null
    let mutable contTkn = firstContTkn
    while (contTkn <> null) do
        let! nextTkn = getBatch contTkn
        contTkn <- nextTkn
```

Sincere apologies, Isaac. I should have spotted this but, more importantly, I should have had this covered by a test. It never occurred to me that I didn't have a multi-batch test.

**Comment by isaacabraham** [comment]
No worries :-) I'll push a fix out - code hasn't been released yet anyway.

**Comment by isaacabraham** [comment]
Would you have time to write a multi-batch test though? :-)

**Comment by stewart-r** [comment]
Certainly - its the least I can do. :-)

I'm just starting work right now but will either do it at lunchtime or this evening

**Comment by isaacabraham** [comment]
That's great - I've pushed the fix, so no rush on the test case. Thanks for your help here :-)

**Pull request merged**